ddl_checker/main.go

```diff
@@ -0,0 +1,214 @@
+		return unicode.IsSpace(r) || r == ';'
+	}))
+	if lowerTrimInput == "exit" {
+		return false
+	}
+	// cmd setmod
+	if strings.HasPrefix(lowerTrimInput, "setmod") {
+		switch strings.TrimSpace(lowerTrimInput[6:]) {
+		case "auto":
+			modCode = 0
+		case "prompt":
+			modCode = 1
+		case "manual":
+			modCode = 2
+		default:
+			fmt.Println("SETMOD usage: SETMOD <MODCODE>; MODCODE = [\"Auto\", \"Prompt\", \"Manual\"] (case insensitive).")
+		}
+		return true
+	}
+	if modCode != 2 {
+		// auto and query mod
+		stmt, err := executableChecker.Parse(input)
+		if err != nil {
+			fmt.Printf("[DDLChecker] SQL parse error: %s\n", err.Error())
+			return true
+		}
+		neededTable := ddl_checker.GetTableNeededExist(stmt)
+		nonNeededTable := ddl_checker.GetTableNeededNonExist(stmt)
+		if modCode == 0 || (modCode == 1 && queryAutoSync(neededTable, nonNeededTable)) {
+			syncTablesFromMysql(neededTable)
+			dropTables(nonNeededTable)
+		}
+	}
+	err := executableChecker.Execute(input)
+	if err == nil {
+		fmt.Println("[DDLChecker] SQL execution succeeded")
+	} else {
+		fmt.Println("[DDLChecker] SQL execution failed:", err.Error())
+	}
+	return true
+}
+
+func syncTablesFromMysql(tableNames []string) {
+	for _, tableName := range tableNames {
+		fmt.Println("[DDLChecker] Syncing Table", tableName)
+		isExist := executableChecker.IsTableExist(tableName)
+		if isExist {
+			fmt.Println("[DDLChecker] Table", tableName, "is exist,Skip")
+			continue
+		}
+		row := db.QueryRow("show create table `" + tableName + "`")
+		var table string
+		var createTableDDL string
+		err := row.Scan(&table, &createTableDDL)
+		if err != nil {
+			fmt.Println("[DDLChecker] SQL Execute Error:", err.Error())
+			continue
+		}
+		err = executableChecker.Execute(createTableDDL)
+		if err != nil {
+			fmt.Println("[DDLChecker] Create table failure:", err.Error())
+			continue
+		}
+	}
+}
+
+func dropTables(tableNames []string) {
+	for _, tableName := range tableNames {
+		fmt.Println("[DDLChecker] Dropping table", tableName)
+		err := executableChecker.Execute(fmt.Sprintf("drop table if exists %s", tableName))
+		if err != nil {
+			fmt.Println("[DDLChecker] DROP TABLE", tableName, "Error:", err.Error())
+		}
+	}
+}
+
+func queryAutoSync(neededTable []string, nonNeededTable []string) bool {
+	for {
+		fmt.Printf("[DDLChecker] Do you want to synchronize table %v from MySQL "+
+			"and drop table %v in DDLChecker?(Y/N)", neededTable, nonNeededTable)
+	innerLoop:
+		for {
+			result, err := reader.ReadString('\n')
+			if err != nil {
+				fmt.Printf("[DDLChecker] Read stdin error: %s\n", err.Error())
+				return false
+			}
+			switch strings.ToLower(strings.TrimSpace(result)) {
+			case "y", "yes":
+				return true
+			case "n", "no":
+				return false
+			case "":
+				continue innerLoop
+			default:
+				break innerLoop
+			}
+		}
+	}
+}
+
+func printWelcome() {
+	fmt.Print(WelcomeInfo)
+}
+
+func modeName() string {
+	switch modCode {
+	case 0:
+		return " Auto "
```

Comment:
why the spaces around "Auto" 🤔 

```suggestion
		return "Auto"
```